src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.CSharp.Tests.cs

```diff
@@ -0,0 +1,229 @@
+        string a = ""aBc"";
+        string b = ""bc"";
+        [|{diagnosedLine}|];
+    }}
+}}";
+            string fixedCode = $@"using System;
+class C
+{{
+    void M()
+    {{
+        string a = ""aBc"";
+        string b = ""bc"";
+        {fixedLine};
+    }}
+}}";
+            await VerifyCSharpAsync(originalCode, fixedCode);
+        }
+
+        [Theory]
+        [MemberData(nameof(DiagnosedAndFixedStringLiteralsData))]
+        public async Task Diagnostic_StringLiterals_ReturnExpressionBody(string diagnosedLine, string fixedLine)
+        {
+            string originalCode = $@"using System;
+class C
+{{
+    object M() => [|{diagnosedLine}|];
+}}";
+            string fixedCode = $@"using System;
+class C
+{{
+    object M() => {fixedLine};
+}}";
+            await VerifyCSharpAsync(originalCode, fixedCode);
+        }
+
+        [Theory]
+        [MemberData(nameof(DiagnosedAndFixedStringReturningMethodsData))]
+        public async Task Diagnostic_StringReturningMethods_Discard(string diagnosedLine, string fixedLine)
+        {
+            string originalCode = $@"using System;
+class C
+{{
+    public string GetStringA() => ""aBc"";
+    public string GetStringB() => ""CdE"";
+    void M()
+    {{
+        _ = [|{diagnosedLine}|];
+    }}
+}}";
+            string fixedCode = $@"using System;
+class C
+{{
+    public string GetStringA() => ""aBc"";
+    public string GetStringB() => ""CdE"";
+    void M()
+    {{
+        _ = {fixedLine};
+    }}
+}}";
+            await VerifyCSharpAsync(originalCode, fixedCode);
+        }
+
+        [Theory]
+        [MemberData(nameof(DiagnosedAndFixedParenthesizedData))]
+        public async Task Diagnostic_Parenthesized_ReturnCastedToString(string diagnosedLine, string fixedLine)
+        {
+            string originalCode = $@"using System;
+class C
+{{
+    string M()
+    {{
+        return ([|{diagnosedLine}|]).ToString();
+    }}
+}}";
+            string fixedCode = $@"using System;
+class C
+{{
+    string M()
+    {{
+        return ({fixedLine}).ToString();
+    }}
+}}";
+            await VerifyCSharpAsync(originalCode, fixedCode);
+        }
+
+        [Theory]
+        [MemberData(nameof(NoDiagnosticContainsData))]
+        [InlineData("\"aBc\".CompareTo(null)")]
+        [InlineData("\"aBc\".ToUpperInvariant().CompareTo((object)null)")]
+        public async Task NoDiagnostic_All(string ignoredLine)
+        {
+            string originalCode = $@"using System;
+class C
+{{
+    object M()
+    {{
+        char ch = 'c';
+        object obj = 3;
+        return {ignoredLine};
+    }}
+}}";
+
+            await VerifyCSharpAsync(originalCode, originalCode);
+        }
+
+        private async Task VerifyCSharpAsync(string originalSource, string fixedSource)
+        {
+            VerifyCS.Test test = new()
+            {
+                TestCode = originalSource,
+                FixedCode = fixedSource,
+                MarkupOptions = MarkupOptions.UseFirstDescriptor
```

Comment:
This shouldn't be needed.

Reply:
The exception message says:

```
Multiple diagnostic descriptors with ID {diagnosticId} were found
```

I interpreted it differently: you have more than one descriptor, they all share the same ID. Which is my case.

Reply:
Indeed. I read it completely incorrectly.